src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { DocumentClient, QueryIterator, CollectionMeta, FeedOptions, ProcedureMeta } from 'documentdb';
+import { DocDBTreeItemBase } from './DocDBTreeItemBase';
+import { IAzureTreeItem } from 'vscode-azureextensionui';
+import { DocDBStoredProcedureTreeItem } from './DocDBStoredProcedureTreeItem';
+
+/**
+ * This class represents the DocumentDB "Stored Procedures" node in the tree
+ */
+export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<ProcedureMeta> {
+    public static contextValue: string = "cosmosDBStoredProceduresGroup";
+    public readonly contextValue: string = DocDBStoredProceduresTreeItem.contextValue;
+    public readonly childTypeLabel: string = "Stored Procedure";
+
+    constructor(documentEndpoint: string, masterKey: string, private _collection: CollectionMeta, isEmulator: boolean) {
+        super(documentEndpoint, masterKey, isEmulator);
+    }
+
+    public initChild(resource: ProcedureMeta): IAzureTreeItem {
+        return new DocDBStoredProcedureTreeItem(resource);
+    }
+
+    public get iconPath(): any {
+        return {
+            light: path.join(__filename, '..', '..', '..', '..', '..', 'resources', 'icons', 'theme-agnostic', 'stored procedures.svg'),
+            dark: path.join(__filename, '..', '..', '..', '..', '..', 'resources', 'icons', 'theme-agnostic', 'stored procedures.svg')
+        };
+    }
+
+    public get id(): string {
+        return "$StoredProcedures";
+    }
+
+    public get label(): string {
+        return "Stored Procedures";
+    }
+
+    public get link(): string {
+        return this._collection._self;
+    }
```

Comment:
I see both `DocDBDocumentsTreeItem` and `DocDBStoredProceduresTreeItem` have the same self-link. If user-defined-functions and triggers are in the pipeline, I'd expect the container treeItem(s) to have the same self-link too (that of the collection). In order to reduce code repetition, would it make sense to have `DocDBDocumentsTreeItem` and `DocDBStoredProceduresTreeItem` inherit this property?

Reply:
It's calling the parent's method, so not really duplicated code. I think this is better than adding it to the super for this small amount.